### PR TITLE
docs: include modern sass vite options

### PR DESCRIPTION
**Pull request by userquin**
## Description

This PR adds a new section in `Caveats` to enable SASS modern Vite options for Vite and Nuxt.

## Markup:

NA

### CODE REVIEW

**Comment by KaelWD** [comment]
https://vitejs.dev/config/shared-options.html#css-preprocessoroptions says to also use sass-embedded

**Comment by userquin** [comment]
> https://vitejs.dev/config/shared-options.html#css-preprocessoroptions says to also use sass-embedded

Then you need to replace `sass`  with `sass-embedded`: check https://github.com/vitejs/vite/pull/17728#issuecomment-2247572134

Will Vuetify work with `sass-embedded`?. IIRC you closed a PR about deprecated mixed-decl: we need to do some tests, I'm not a sass expert.

**Comment by KaelWD** [comment]
I hadn't seen the PR and it didn't link the issue so I fixed it myself: #20139

They used our code in that test so I don't see why it wouldn't work.

File: packages/docs/src/pages/en/features/sass-variables.md [hunk 1]
```diff
@@ -273,10 +273,47 @@ Only put variables, mixins, and functions in the settings file, styles should be
 Vuetify loads precompiled CSS by default, enabling variable customization will switch to the base SASS files instead which must be recompiled with your project.
 This can be a performance hit if you're using more than a few vuetify components, and also forces you to use the same SASS compiler version as us.
 
+If you're using Vite or Nuxt 3 (with Vite), check [sass modern compiler](#sass-modern-compiler) to improve performance.
```
Comment (KaelWD) [inline packages/docs/src/pages/en/features/sass-variables.md:276]: I don't think this needs a whole separate section.

```suggestion
Performance can be improved with Vite by using the modern sass compiler. Replace your `sass` dependency with `sass-embedded`, update vite to 5.4 or later, and set [`css.preprocessorOptions.sass.api`](https://vitejs.dev/config/shared-options#css-preprocessoroptions) to `'modern-compiler'` in the vite config.
```

Comment (userquin) [reply packages/docs/src/pages/en/features/sass-variables.md:276]: so, `sass` not required anymore? we should remove it from docs

Comment (userquin) [reply packages/docs/src/pages/en/features/sass-variables.md]: done

**Review by KaelWD — APPROVED** [review]

**Pull request merged**